File: Algorithms/01-Warmup/07-min-max-sum/07.py
#!/usr/bin/env python


def miniMaxSum(arr):
    lower = 0
    higher = 0
    for i in range(0, len(arr)):
        mysum = 0
        for j in range(0, len(arr)):
            if i != j:
                mysum += arr[j]
        if i == 0 or lower > mysum:
            lower = mysum
        if mysum > higher:
            higher = mysum
    print(lower, higher)


if __name__ == "__main__":
    arr = list(map(int, input().strip().split(' ')))
    miniMaxSum(arr)

File: Algorithms/01-Warmup/07-min-max-sum/07.rb
#!/usr/bin/env ruby

def mini_max_sum(arr)
    higher = 0
    lower = 0
    (0...arr.length).each do |i|
        sum = 0
        (0...arr.length).each do |j|
            sum += arr[j] if i != j
        end
        lower = sum if i.zero? || lower > sum
        higher = sum if sum > higher
    end
    puts "#{lower} #{higher}"
end

arr = gets.strip
arr = arr.split(" ").map(&:to_i)
mini_max_sum(arr)
